src/main/scala/org/broadinstitute/dsde/agora/server/business/AgoraBusiness.scala

```diff
@@ -27,21 +26,38 @@ object AgoraBusiness {
     entityWithId.copy(url = Option(agoraUrl(entityWithId)))
   }
 
+  def filterByReadPermissions(authEntities: Seq[AuthorizedAgoraEntity]): Seq[AgoraEntity] = {
+    authEntities.flatMap(authEntity => filterByReadPermissions(authEntity))
+  }
+
+  def filterByReadPermissions(authEntity: AuthorizedAgoraEntity): Option[AgoraEntity] = {
+    if (authEntity.authorization.canRead) Some(authEntity.entity) else None
+  }
+
   def find(agoraSearch: AgoraEntity,
            agoraProjection: Option[AgoraEntityProjection],
            entityTypes: Seq[AgoraEntityType.EntityType]): Seq[AgoraEntity] = {
-    AgoraDao.createAgoraDao(entityTypes).find(agoraSearch, agoraProjection).map(entity => entity.copy(url = Option(agoraUrl(entity))))
+    val entities = AgoraDao.createAgoraDao(entityTypes).find(agoraSearch, agoraProjection).map(entity => entity.copy(url = Option(agoraUrl(entity))))
+    filterByReadPermissions(authorizationProvider.authorizationsForEntities(entities))
   }
 
   def findSingle(namespace: String,
                  name: String,
                  snapshotId: Int,
                  entityTypes: Seq[AgoraEntityType.EntityType]): Option[AgoraEntity] = {
-    AgoraDao.createAgoraDao(entityTypes).findSingle(namespace, name, snapshotId).map(entity => entity.copy(url = Option(agoraUrl(entity))))
+    val optEntity = AgoraDao.createAgoraDao(entityTypes).findSingle(namespace, name, snapshotId).map(entity => entity.copy(url = Option(agoraUrl(entity))))
```

Comment:
The following seems like it would save you the map, and just move the entityCopy within the check. Seems like we're double-checking for None by doing both. I was hoping it would be /less/ cluttered. But looking at it, it's probably 6 of one...
```
    val optEntity = AgoraDao.createAgoraDao(entityTypes).findSingle(namespace, name, snapshotId)
    optEntity match {
      case Some(foundEntity) => filterByReadPermissions(authorizationProvider.authorizationsForEntity(foundEntity.copy(url = Option(agoraUrl(foundEntity)))))
      case None => None
    }
  }
```